resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v1/LoggingPodStatusWatcher.scala

```diff
@@ -44,15 +44,18 @@ private[kubernetes] class LoggingPodStatusWatcher(podCompletedFuture: CountDownL
   private val logRunnable: Runnable = new Runnable {
     override def run() = logShortStatus()
   }
-  if (interval > 0) {
-    scheduler.scheduleWithFixedDelay(logRunnable, 0, interval, TimeUnit.MILLISECONDS)
-  }
 
   private var pod: Option[Pod] = Option.empty
   private def phase: String = pod.map(_.getStatus().getPhase()).getOrElse("unknown")
   private def status: String = pod.map(_.getStatus().getContainerStatuses().toString())
     .getOrElse("unknown")
 
+  def start(): Unit = {
+    if (interval > 0) {
+      scheduler.scheduleWithFixedDelay(logRunnable, 0, interval, TimeUnit.MILLISECONDS)
```

Comment:
I think a more direct fix for this LoggingPodStatusWatcher holding the JVM open issue is to make sure that the scheduleThreadPool is launching daemon threads vs non-daemon threads.

That way we don't need to introduce this additional start method and can leave the scheduler start as a side effect of the constructor (and other scenarios where the scheduler fails to get stopped won't hold the JVM open)

What if `scheduler = ThreadUtils.newDaemonSingleThreadExecutor("logging-pod-status-watcher")` ?